vignettes/Tutorial.Rmd

```diff
@@ -0,0 +1,103 @@
+
+This notebook shows you how 
+1. Specify any MGF file
+2. Run a library search through the Spec2Vec API
+3. Save the results as CSV.
+
+# 1. (Down)load a MS/MS dataset (MGF format)
+
+We'll use a small MS/MS dataset in the MGF format from [here](https://gnps-external.ucsd.edu/gnpslibrary/GNPS-COLLECTIONS-MISC.mgf).
+You can also select any other from the [GNPS spectral library](https://gnps-external.ucsd.edu/gnpslibrary), or - of course - use your own.
+
+*Note that you will need the Precursor_MZ field `PEPMASS` and the abundance pairs in your MGF file.*
+
+```{r}
+# Load your own MS/MS dataset (and skip the next cell)
+path_to_mgf <- '/path/to/local_dataset.mgf'
+```
+
+```{r}
+# OR download a small MS/MS dataset from GNPS, in the same directory as this notebook
+url = 'https://gnps-external.ucsd.edu/gnpslibrary/GNPS-COLLECTIONS-MISC.mgf'
+
+path_to_mgf = 'GNPS-COLLECTIONS-MISC.mgf' # use your prefered saving path here
+
+download.file(url, path_to_mgf, method = "curl")
+```
+
+# 2. Query for the best matches with Spec2Vec
+
+`Romigami` is a R wrapper which instantiates a virtual environment on which to install and call the Python `Omigami` package
+
+`Omigami` is a Python wrapper which :
+- Builds a json payload from the MGF file
+- Calls the Spec2Vec API
+- Formats the prediction results into readable dataframes
+
+____
+- `n_best` sets the number of matches you'd like per spectrum (it is set to 10 by default).
+
+In the results dataframes, the input spectra can be identified by their number in the dataframes index, which refers to their order in the MGF file.  
+*i.e.* `matches of spectrum 1` gives the spectrum_id and Spec2Vec scores of the library spectra matches, for the first spectrum in the MGF file.
+
+For each spectrum in the MGF file, the library spectra matches are sorted according to their Spec2Vec similarity score (best is first).   
+The following information about the predicted spectra are returned :
+- `score`, the Spec2Vec similarity score between the input spectrum and the library spectrum
+- `matches of spectrum #`, the spectrum_ID of the matched library spectra for the spectrum number # in the MGF file
+
+```{r}
+# import the package
+library("romigami")
+```
+
+```{r}
+# environment setup
+initialize_environment()
+```
+
+```{r}
+# Run Spec2Vec library search with your user token
+spectra_matches <- match_spectra_from_path(token = "8DmWwNrFB8oSQcjHsEabHoNcUy30MZLY",
+                        mgf_path = path_to_mgf,
+                        n_best = 10)
+```
+
+## 2.1 View results
+A list of dataframes is returned. To look at a specific dataframe you can call :
+```python
+spectra_matches[i]  # 'i' refers to the ith spectrum in the MGF file input
+```
+
+```{r}
+spectra_matches[1]
+```
+
+# 3. Save results
+
+Execute the following cell to save the results in a CSV file. For readability, each dataframe is saved in its own CSV file under the `matches` directory.
```

Comment:
Here it should be 'cell' or 'call' ?

Reply:
It should be 'cell'. 
Rmd is like a notebook but for R, so it has 'cells'.